prime-router/src/main/kotlin/Schema.kt

```diff
@@ -1,17 +1,50 @@
 package gov.cdc.prime.router
 
-// A PRIME schema contains a collection of data elements, mappings, validations and standard algorithms
-// needed to translate data to the form that a public health authority needs
-//
+/**
+ * A PRIME schema contains a collection of data elements, mappings, validations and standard algorithms
+ * needed to translate data to the form that a public health authority needs.
+ *
+ * Schemas can be either be `basedOn` another schema or `extend` another schema. If
+ * a schema is `basedOn` another schema, the elements of the schema that match the
+ * the elements of the base schema inherit their properties. The `basedOn` feature is
+ * useful when defining a state CSV schema because it allows the state schema to only specify the names
+ * of the elements it wants and the unique properties of that element. For example:
+ *
+ * <code>
+ * - name: state_schema
+ *   basedOn: covid-19 # means all elements inherit properties from this schema
+ *   elements:
+ *   - name: patient_first_name
+ *     csvFields: [{name: PatientFirstName }] # adds a property that is not in the base schema
+ * </code>
+ *
+ * Schemas can also have a `extends` relationship with another schema. Like a `basedOn` relationship,
+ * the elements of a schema inherit from the `extends` schema. In addition, all the elements of the `extends`
+ * schema are appended to the schema. The `extends` relationship is useful in building HL7 schemas, because
+ * these schema mostly have a common element set.
+ *
+ * @param name is the unique name of the schema
+ * @param topic defines the set of schemas that can translate with each other
+ * @param elements are the data elements of the schema
+ * @param trackingElement is the element that is unique per item and is used for tracking
+ * @param description is the text description of the schema
+ * @param referenceUrl is a link to a standard document used to create the schema
+ * @param extends is a name of a schema that this schema extends.
+ * @param basedOn is a name of a schema that this schema is based on
+ */
 data class Schema(
```

Comment:
JavaDoc Comments. What a concept.

Reply:
At a summer job, I wrote an auto-documentation generator based on an invented "code decorating language" that years later turned out to be remarkably similar to javadoc.